File: ExampleCodes/Javascript/Exponential.js
console.log("Starting...");

const Exponential = (k, n) => {
  //O(k)^n complexity or in this case O(k)5 = 32 operations
  for (let i = 0; i < Math.pow(k, n); i++) {
    console.log("i is " + i);
  }
};

const { performance } = require("perf_hooks");
const time1 = performance.now();
Exponential(2, 5); //this is n
const time2 = performance.now();
console.log(`Elapsed time: ${(time2 - time1) / 1000} seconds`);

File: ExampleCodes/rust/exponential/src/exponential.rs
use num_traits::Pow;
use std::time::Instant;
pub fn exponential() {
    println!("Starting ... ");
    let k: u32 = 2;
    let n: u32 = 5;
    Pow::pow(k, n);
    //k^n so to to power of 5 = 32 operations

    let now = Instant::now(); //records this instant

    //Constant Time Complexity Example
    for x in 0..Pow::pow(k, n) {
        println!("{}", x);
    }

    let new_now = Instant::now();

    print!("Took ");
    print!("{:?}", new_now.checked_duration_since(now));
    print!(" To Complete");
}
